### DISCUSSION TITLE
Streaming data

### DISCUSSION THREAD

**Discussion by CBaerentsen** [tdorssers/TeslaPy#110] [Q&A]
I am programming my car such a raspberry pi releases the charging cable when the trunk is open. The problem is that I am asking for vehicle data every 4 seconds. 

I don't quite understand how to stream the data such I will be waiting for an event to happen, instead of asking every 4 seconds if something has changed.

Is streaming the data or waiting for an event change possible?

Many thanks!

**Comment by tdorssers** [comment]
You can't use streaming data to wait for an event. Streaming data stops if values are unchanged for 10 seconds, so it's most useful while driving. If you are waiting for an event, you need to poll the API. The provided gui.py polls every 5 seconds for instance.
